### PR TITLE
Validation of BMH NetworkData

### PR DESCRIPTION
**Pull request by alexander-hughes**
Signed-off-by: Alexander Hughes <Alexander.Hughes@pm.me>

### CODE REVIEW

File: pkg/vbmh/machines.go [hunk 1]
```diff
@@ -97,15 +97,21 @@ func (m *Machine) String() string {
 	return fmt.Sprintf("Machine {\n\tBmh:%s\n\tScheduleStatus:%s\n\tVmRole:%v\n}\n", m.BMH.ObjectMeta.Name, m.ScheduleStatus, m.VMRole)
 }
 
-func NewMachine(bmh metal3.BareMetalHost, nodeRole airshipv1.VmRoles, schedState ScheduledState) (m *Machine) {
+func NewMachine(bmh metal3.BareMetalHost, nodeRole airshipv1.VmRoles, schedState ScheduledState) (m *Machine, e *ErrorNetworkDataNotFound) {
```
Comment (teoyaomiqui) [inline pkg/vbmh/machines.go]: i think return signature should be (m *Machine, e error) instead

**Review by teoyaomiqui — CHANGES_REQUESTED** [review]

File: pkg/vbmh/machines.go (hunk 1 above)
Comment (alexander-hughes) [reply pkg/vbmh/machines.go]: done

File: config/crd/bases/airship.airshipit.org_sipclusters.yaml [hunk 2]
```diff
@@ -4,7 +4,7 @@ apiVersion: apiextensions.k8s.io/v1beta1
 kind: CustomResourceDefinition
 metadata:
   annotations:
-    controller-gen.kubebuilder.io/version: (devel)
+    controller-gen.kubebuilder.io/version: v0.2.5
```
Comment (drewwalters96) [inline config/crd/bases/airship.airshipit.org_sipclusters.yaml]: Changes to this file revert an earlier commit: https://github.com/att-comdev/sip/commit/0f24ea964cb0e43393039ac3813e3018cdad2a36

File: pkg/vbmh/machines.go [hunk 3]
```diff
@@ -289,7 +295,12 @@ func (ml *MachineList) countScheduledAndTobeScheduled(nodeRole airshipv1.VMRoles
 		if readyScheduled {
 			logger.Info("BMH host is not yet marked as ready to be scheduled, marking it as ready to be scheduled")
 			// Add it to the list.
-			ml.Machines[bmh.ObjectMeta.Name] = NewMachine(bmh, nodeRole, Scheduled)
+			m, err := NewMachine(bmh, nodeRole, Scheduled)
+			if err != nil {
+				logger.Info("Skipping BMH host as it did not meet creation requirements", "error", err.Error())
```
Comment (drewwalters96) [inline pkg/vbmh/machines.go]: From the user's perspective, they have already created machines. I would recommend changing this to "did not meet scheduling requirements"

File: pkg/vbmh/vbmh_test.go [hunk 4]
```diff
@@ -25,7 +25,7 @@ var _ = Describe("MachineList", func() {
 		nodes := map[string]*Machine{}
 		for n := 0; n < numNodes; n++ {
 			bmh, _ := testutil.CreateBMH(n, "default", "master", 6)
-			nodes[bmh.Name] = NewMachine(*bmh, airshipv1.VMMaster, NotScheduled)
+			nodes[bmh.Name], _ = NewMachine(*bmh, airshipv1.VMMaster, NotScheduled)
```
Comment (drewwalters96) [inline pkg/vbmh/vbmh_test.go]: Here we probably want to do `Expect(err).To(BeNil())`

**Review by drewwalters96 — CHANGES_REQUESTED** [review]
This looks good, Alex. Can we also add a test case to verify that this handles a missing network data secret reference?

File: pkg/vbmh/vbmh_test.go (hunk 4 above)
Comment (alexander-hughes) [reply pkg/vbmh/vbmh_test.go]: done.

File: pkg/vbmh/machines.go (hunk 3 above)
Comment (alexander-hughes) [reply pkg/vbmh/machines.go]: done

File: config/crd/bases/airship.airshipit.org_sipclusters.yaml (hunk 2 above)
Comment (alexander-hughes) [reply config/crd/bases/airship.airshipit.org_sipclusters.yaml]: reverted file back to 0f24ea964cb0e43393039ac3813e3018cdad2a36

**Review by drewwalters96 — CHANGES_REQUESTED** [review]
Thanks. Can we also add a test case to verify that this handles a missing network data secret reference?

**Comment by drewwalters96** [comment]
It would also be much appreciated if you could tidy the commit history before we merge

**Comment by alexander-hughes** [comment]
> 
> 
> Thanks. Can we also add a test case to verify that this handles a missing network data secret reference?

Test case added in latest commit, will squash everything down once the patch doesn't have anymore requested changes.

File: config/manager/kustomization.yaml [hunk 5]
```diff
@@ -1,2 +1,7 @@
 resources:
 - manager.yaml
+apiVersion: kustomize.config.k8s.io/v1beta1
+kind: Kustomization
+images:
+- name: controller
+  newName: quay.io/airshipit/sip
```
Comment (teoyaomiqui) [inline config/manager/kustomization.yaml:7]: We should remove this as well

**Pull request closed without merging**